src/app/common/components/Prompt.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import { Modal, Button } from '@patternfly/react-core';
+import { useHistory } from 'react-router';
+import { noop } from '@app/utils/utils';
+import ConfirmModal from '@app/common/components/ConfirmModal';
+
+interface IPrompt {
+  when: boolean;
+  title: string;
+  message: string | JSX.Element;
+  okText?: string;
+  cancelText?: string;
+}
+
+const Prompt: React.FunctionComponent<IPrompt> = ({
+  when,
+  title,
+  message,
+  okText = 'OK',
+  cancelText = 'Cancel',
+}) => {
+  const history = useHistory();
+  const [showPrompt, setShowPrompt] = React.useState(when);
+  const [currentPath, setCurrentPath] = React.useState('');
+
+  React.useEffect(() => {
+    if (when) {
+      history.block((prompt) => {
+        setCurrentPath(prompt.pathname);
+        setShowPrompt(true);
+        return 'true';
+      });
+    }
+  }, [history, when]);
```

Comment:
This `useEffect` successfully sets up the block when `when` becomes true, but if `when` becomes false again it doesn't remove the block. That doesn't matter much for this use case, but the prior behavior is that you're only blocked if some form is dirty (e.g. I can open a plan for editing, step through it without changing anything, and close it with no warning). I think if this will be reusable in lib-ui it should also clean up this block if `when` becomes false.

Looking at [this stackoverflow answer](https://stackoverflow.com/a/65527222) it seems like the cleanest way to do that might be to have the effect always call `history.block` with `if (when)` inside its callback, and then have the effect return a cleanup function that calls `unblock` (stored from the return value of `history.block()`). That way the block is always in place as long as the component is mounted but it checks the `when` condition every time it is hit.

Reply:
Ok, yea that makes sense. I was trying to find more info on how to best utilize that return function, and had actually gone down that road a bit but I wasn't quite sure so I left it minimal for that first pass. I'll tidy this up  👍